app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/blocklist/AppTPBlockListInterceptorApiPlugin.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2025 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.vpn.blocklist
+
+import com.duckduckgo.app.global.api.ApiInterceptorPlugin
+import com.duckduckgo.common.utils.plugins.pixel.PixelParamRemovalPlugin
+import com.duckduckgo.common.utils.plugins.pixel.PixelParamRemovalPlugin.PixelParameter
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.feature.toggles.api.FeatureTogglesInventory
+import com.duckduckgo.mobile.android.vpn.AppTpVpnFeature
+import com.duckduckgo.mobile.android.vpn.VpnFeaturesRegistry
+import com.duckduckgo.mobile.android.vpn.feature.AppTpRemoteFeatures.Cohorts.CONTROL
+import com.duckduckgo.mobile.android.vpn.feature.AppTpRemoteFeatures.Cohorts.TREATMENT
+import com.duckduckgo.mobile.android.vpn.feature.activeAppTpTdsFlag
+import com.duckduckgo.mobile.android.vpn.pixels.DeviceShieldPixelNames
+import com.duckduckgo.mobile.android.vpn.pixels.DeviceShieldPixels
+import com.squareup.anvil.annotations.ContributesMultibinding
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import javax.inject.Inject
+import kotlinx.coroutines.runBlocking
+import logcat.logcat
+import okhttp3.Interceptor
+import okhttp3.Interceptor.Chain
+import okhttp3.Response
+import retrofit2.Invocation
+
+@ContributesMultibinding(
+    scope = AppScope::class,
+    boundType = ApiInterceptorPlugin::class,
+)
+class AppTPBlockListInterceptorApiPlugin @Inject constructor(
+    private val inventory: FeatureTogglesInventory,
+    private val moshi: Moshi,
+    private val pixel: DeviceShieldPixels,
+    private val vpnFeaturesRegistry: VpnFeaturesRegistry,
+) : Interceptor, ApiInterceptorPlugin {
+
+    private val jsonAdapter: JsonAdapter<Map<String, String>> by lazy {
+        moshi.adapter(Types.newParameterizedType(Map::class.java, String::class.java, String::class.java))
+    }
+    override fun intercept(chain: Chain): Response {
+        val request = chain.request().newBuilder()
+
+        val tdsRequired = chain.request().tag(Invocation::class.java)
+            ?.method()
+            ?.isAnnotationPresent(AppTPTdsRequired::class.java) == true
+
+        val appTpEnabled = runBlocking {
+            vpnFeaturesRegistry.isFeatureRunning(AppTpVpnFeature.APPTP_VPN)
```

Comment:
This is a private API, I asked Aitor and you should use https://duckduckgo.github.io/Android/app-tracking-api/com.duckduckgo.mobile.android.app.tracking/-app-tracking-protection/index.html#-507964147%2FFunctions%2F1663220899 instead which returns true if AppTP is enabled and also running. If you only want to know if it's enabled, then use isEnabled() but always from the public API :)